app/models/manager_refresh/inventory_collection_default/infra_manager.rb

```diff
@@ -42,8 +57,8 @@ def guest_devices(extra_attributes = {})
     def hardwares(extra_attributes = {})
       attributes = {
         :model_class => ::Hardware,
-        :manager_ref => [:vm_or_template],
-        :association => :hardwares
+        :manager_ref => [:vm_or_template], # :host
```

Comment:
So I think the best way to handle this (@ladas can chime in if he has a better way) would be to have two associations on `:ext_management_system` one for `vm_hardwares` and the other for `host_hardwares` that way you can use the right manager ref for each one